src/components/Thought.tsx

```diff
@@ -323,6 +327,15 @@ const ThoughtContainer = ({
   const styleContainer = useThoughtStyleContainer({ children, env, styleContainerProp, thoughtId, path })
   const value = useSelector(state => getThoughtById(state, thoughtId)?.value)
 
+  // Determine if the thought contains a URL that should be ellipsized
+  // URLs should not be treated as multiline to prevent unwanted line breaks
+  const ellipsizedUrl = !isEditing && containsURL(value || '')
+
+  // Detect if the thought content spans multiple lines
+  // This hook handles timing issues and provides both immediate and delayed calculations
+  // to prevent layout flickering while ensuring accurate measurements
```

Comment:
These are implementation details that the Thought component does not need to know about. For encapsulation purposes, it's better to keep those details contained in the custom hook unless the parent component needs to know about them.